storage/shard.go

```diff
@@ -906,18 +975,37 @@ func (s *dbShard) tryRetrieveWritableSeries(id ident.ID) (
 	return nil, opts, nil
 }
 
-func (s *dbShard) newShardEntry(id ident.ID) *dbShardEntry {
+func (s *dbShard) newShardEntry(id ident.ID, tags ident.TagIterator) (*dbShardEntry, error) {
+	clonedTags, err := s.cloneTags(tags)
```

Comment:
Why do the tags need to be cloned? Can we just not finalize them with the context?

Reply:
same reason we need clone the ID (two lines below this): the lifecycle of the underlying bytes is tied to the context which is finalised after the server side RPC response. Need to clone to have manage the lifecycle independently.

Reply:
Yeah it just seems like we could not tie the lifecycle of the bytes to the context (and just let them leak in the rare error case) but I understand that this is safer

Reply:
The full write path allocation is a little sketchier than my original comment lets on: the bytes underlying the ident's here (in `newShardEntry`) are actually allocated by tchannel, we wrap them up into checked.Bytes (and then an ident) using a CheckedBytesWrapperPool in the m3db server side code. The context finalizing just releases the references the ident.ID and the checked.Bytes have to the underlying tchannel bytes slice.

Reply:
Going to stick to the simple version of cloning (as we do now), and enhance in subsequent PRs based on benchmarks (as needed)